File: NPC AI work/Finite State Machine/EnemyAI/States/EStatePatrol.cpp
// Fill out your copyright notice in the Description page of Project Settings.


#include "EnemyAI/States/EStatePatrol.h"

// Sets default values for this component's properties
UEStatePatrol::UEStatePatrol()
{

}

void UEStatePatrol::OnEnter()
{
	UE_LOG(LogTemp, Warning, TEXT("Patrol Entered"));
}

void UEStatePatrol::SMTick()
{
	UE_LOG(LogTemp, Warning, TEXT("Patrol Ticked"));
}

void UEStatePatrol::OnExit()
{
	UE_LOG(LogTemp, Warning, TEXT("Patrol Left"));
}

File: NPC AI work/Finite State Machine/C# Implementation/EStatePatrol.cs
using System;
using UnityEngine;
using UnityEngine.AI;

public class EStatePatrol : IState
{
    Enemy_Basic parent;

    public EStatePatrol(Enemy_Basic parent)
    {
        this.parent = parent;
    }
    public void OnEnter()
    {
        Debug.Log("Patrol Entered");
    }

    public void OnExit()
    {
        Debug.Log("Patrol Exited");
    }

    public void Tick()
    {
        Debug.Log("Patrol Ticked");

    }

}